client/actions/cohort.js

```diff
@@ -0,0 +1,20 @@
+import {
+    COHORT_REQUEST_LIST,
+    COHORT_REQUEST_LIST_ERROR,
+    COHORT_REQUEST_LIST_SUCCESS
+} from './types';
+
+export const cohortRequestList = params => async dispatch => {
```

Comment:
this is a "thunk" action - note that its a function that returns an async function, which takes "dispatch" for further dispatching.

Inside it dispatches a "I requested", "error" and "success" states

Reply:
It is kind of similar to a generator. You'll dispatch to yield information out of the thunk. In a more programming generic sense we can call this a coroutine.